### PR TITLE
Ports cleaning system refactor

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/52477
- https://github.com/tgstation/tgstation/pull/55805

## About The Pull Request

This PR centralizes a lot of cleaning code into a singular var on the atom level, wash, instead of snowflake and copypasting cleaning code.

This is, effectively, the gist of it. If you want more details, quotes are below:

> - The washed atom proc has been renamed to wash since you usually name member functions in the base form and not past tense.
> - Attempted to organize the way cleaning operates. In my opinion, cleaning is done in two manners, either by mopping which would affect any floor related crap, or by spraying which would affect anything on the tile.
> - Renamed the is_cleanable macro to ismopable to better define this difference. This macro now checks the atom layer instead of using arbitrary item paths
> - When a wash is called on a turf, it will wash off any mopables
> - Better defined the behaviour of cleaning machines (such as janicart). They now clean the turf and any mopables, as well as any small, non-equipped items on the ground, as well as any humans that are lying down.
> - Space cleaner, shower and water vapour are "spraying" cleaning types, they clean both floor decals and mobs.
> -Redesigned the CLEAN_ consts. The new way is more dynamic, cleaning apparatuses can clean more specific than just a cleaning strength.
> - CLEAN_TYPE_* flags indicate a specific cleanable, such as blood, fingerprints or disease
> - CLEAN_* consts consist of a combination of cleaning types to make cleaning apparatuses have a consistent behaviour on what they clean
> - Fixed the shower not properly removing radiation
> - Removed the "washer" argument to the wash proc since it wasn't used anywhere and I don't see when it would ever be used
> - Previously, the COMSIG_COMPONENT_CLEAN_ACT signal would be called manually when you want to clean shit, now you call the wash proc instead and let that call the signal.
> - The COMSIG_COMPONENT_CLEAN_ACT signal is now only used for registration in components. Other custom wash behaviour of objects are done by overriding the wash proc instead.
> - Heavily cleaned up washing of carbon mobs. This was previously performed at multiple places in different ways. Now, /mob/living/carbon overrides the wash proc which makes it clean any non-obscured equipped items. - /mob/living/carbon/human further overrides the wash proc to clean up anything human specific. The space cleaner and cleaning machines now simply call the wash proc on mobs.
> - Added a atom radiate VV verb which makes the object radioactive, used for debugging
> Other less relevant things that I did which I thought were necessary:
> 
> - Moved the bloody_hands var from /mob to /mob/living/carbon/human since no other code used it anyway
> - Fixed the atom_colours list being unnecessarily initialized

The second pr just fixes a bug the first introduced.

## Why It's Good For The Game

Centralizes a major system to be much more forgiving when coding, and not overlap in so many places.

Additionally, I need this for Waterworks, the Plumbing expansion.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Cleaning with spray

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/55f453c9-5f5e-4694-86f4-4c2224089e2b

Spreading viscera works as expected. Either to tiles, or to items/clothing

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/b5e3913a-729c-4917-a717-3d03f35545ce

</details>

## Changelog
:cl: rkz, Donkie, Seris02
refactor: refactored the cleaning system
tweak: adjusted some cleaning intensity values (how well certain things can clean), but not to the point it changes balance
fix: Fixed mobs/items/turfs not being predictably washed by the various methods on board
fix: Fixed showers not properly reducing clothing radioactivity
admin: added a VV to directly add/edit a mob's radioactive green aura.
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
why did every time we want to clean a person, we would throw in like 8 different if statements to check for obscuration. I swear to god

File: code/datums/components/thermite.dm [hunk 1]
```diff
@@ -80,10 +80,10 @@
 	qdel(src)
 
 /datum/component/thermite/proc/clean_react(datum/source, strength)
-	SIGNAL_HANDLER
```
Comment (itsmeow) [inline code/datums/components/thermite.dm]: Put it back, this is a signal handler

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -451,11 +451,9 @@
...
 			things_to_clear += mob_occupant
 			things_to_clear += mob_occupant.GetAllContents()
-		for(var/atom/movable/AM in things_to_clear) //Scorches away blood and forensic evidence, although the SSU itself is unaffected
-			SEND_SIGNAL(AM, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_STRONG)
-			var/datum/component/radioactive/contamination = AM.GetComponent(/datum/component/radioactive)
-			if(contamination)
-				qdel(contamination)
+		for(var/am in things_to_clear) //Scorches away blood and forensic evidence, although the SSU itself is unaffected
+			var/atom/movable/dirty_movable = am
+			dirty_movable.wash(CLEAN_ALL)
```
Comment (itsmeow) [inline code/game/machinery/suit_storage_unit.dm:456]: One thing I don't like about this thing: it has a half-exploit where you can shove someone in here and decontaminate to instantly remove all radiation from them.

File: code/game/objects/items/mop.dm [hunk 3]
```diff
@@ -29,12 +29,12 @@
...
 	return ..()
 
-/obj/item/mop/proc/clean(turf/A)
+/obj/item/mop/proc/clean(turf/A, mob/living/cleaner)
 	if(reagents.has_reagent(/datum/reagent/water, 1) || reagents.has_reagent(/datum/reagent/water/holywater, 1) || reagents.has_reagent(/datum/reagent/consumable/ethanol/vodka, 1) || reagents.has_reagent(/datum/reagent/space_cleaner, 1))
-		SEND_SIGNAL(A, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_MEDIUM)
-		for(var/obj/effect/O in A)
-			if(is_cleanable(O))
-				qdel(O)
+		//SKILLCHIPS, should we want them
```
Comment (itsmeow) [inline code/game/objects/items/mop.dm]: remove the skillchip comment fragments. They can be ported if it gets ported.

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/machinery/suit_storage_unit.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/game/machinery/suit_storage_unit.dm:456]: Hm? I thought that was sort of the purpose of the decontamination mode on these? 

It decontaminates and clears all grime, including radioactive particles from whatever is inside, with it burning the shit out of someone if theres an occupant.

It is sort of an established use, with an exchange of rads for burns.

File: code/game/objects/items/mop.dm (hunk 3 above)
Comment (Tsar-Salat) [reply code/game/objects/items/mop.dm]: addressed in 84335e9

File: code/datums/components/thermite.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/datums/components/thermite.dm]: fixed in 84335e9

**Review by itsmeow — APPROVED** [review]

**Pull request merged**